omp/__init__.py

```diff
@@ -28,9 +28,14 @@ def __init__(self):
         paths = os.environ.get('LD_LIBRARY_PATH', '').split(':')
         for gomp in ('libgomp.so', 'libgomp.dylib'):
             cmd = [cxx, '-print-file-name=' + gomp]
-            path = os.path.dirname(check_output(cmd).strip())
-            if path:
-                paths.append(path)
+            # the subprocess can fail in various ways
+            # in that case just give up that path
+            try:
+                path = os.path.dirname(check_output(cmd).strip())
+                if path:
+                    paths.append(path)
+            except:
```

Comment:
On Tue, Feb 17, 2015 at 10:35:40AM -0800, pbrunet wrote:
> > @@ -28,9 +28,14 @@ def __init__(self):
> >          paths = os.environ.get('LD_LIBRARY_PATH', '').split(':')
> >          for gomp in ('libgomp.so', 'libgomp.dylib'):
> >              cmd = [cxx, '-print-file-name=' + gomp]
> > -            path = os.path.dirname(check_output(cmd).strip())
> > -            if path:
> > -                paths.append(path)
> > +            # the subprocess can fail in various ways
> > +            # in that case just give up that path
> > +            try:
> > +                path = os.path.dirname(check_output(cmd).strip())
> > +                if path:
> > +                    paths.append(path)
> > +            except:
> 
> From pep8:
> 
> ```
> When catching exceptions, mention specific exceptions whenever possible instead of using a bare except: clause.
> ```
> ```
> A bare except: clause will catch SystemExit and KeyboardInterrupt exceptions, making it harder to interrupt a program with Control-C, and can disguise other problems. If you want to catch all exceptions that signal program errors, use except Exception: (bare except is equivalent to except BaseException: ).
> ```

Thanks for the quote! I'll fix this.

> What will happen in Pythran if user explicitly ask for -fopenmp but doesn't support it?

-fopenmp is only triggered at c++ compile time so it's likely the C++
compiler will grunt.